pkg/quarterdeck/auth.go

```diff
@@ -1,20 +1,32 @@
 package quarterdeck
 
 import (
-	"database/sql"
+	"context"
 	"net/http"
+	"time"
 
 	"github.com/gin-gonic/gin"
 	"github.com/golang-jwt/jwt/v4"
 	"github.com/rotationalio/ensign/pkg/quarterdeck/api/v1"
-	"github.com/rotationalio/ensign/pkg/quarterdeck/db"
+	"github.com/rotationalio/ensign/pkg/quarterdeck/db/models"
 	"github.com/rotationalio/ensign/pkg/quarterdeck/passwd"
 	"github.com/rotationalio/ensign/pkg/quarterdeck/tokens"
 	"github.com/rs/zerolog/log"
 )
 
-// TODO: add documentation
+const (
+	DefaultRole = "Member"
+)
+
+// Register creates a new user in the database with the specified password, allowing the
+// user to login to Quarterdeck. This endpoint requires a "strong" password and a valid
+// register request, otherwise a 400 reply is returned. The password is stored in the
+// database as an argon2 derived key so it is impossible for a hacker to get access to
+// raw passwords. By default the user is given the Member role, unless an organization
+// is being created for the user, in which case the user is assigned the Owner role.
+// TODO: add rate limiting to ensure that we don't get spammed with registrations
 // TODO: review and ensure the register methodology is what we want
+// TODO: handle organizations and invites (e.g. with role association).
```

Comment:
It will probably be more than one story - we have to figure out the workflow of user invitations and when organizations are created for users. We haven't discussed this yet, but we probably should.

Reply:
The user invitations workflow came up during a discussion I had with Cletus yesterday, while going through the Ensign Figma designs, so we should touch base on this.

Reply:
Ok, could you schedule a call with Edwin about this tomorrow morning?